esphome/components/online_image/bmp_image.cpp

```diff
@@ -13,21 +13,42 @@ static const char *const TAG = "online_image.bmp";
 
 int HOT BmpDecoder::decode(uint8_t *buffer, size_t size) {
   size_t index = 0;
-  this->decoded_bytes_ += size;
-  if (this->current_index_ == 0 && this->decoded_bytes_ > 14) {
+  if (this->current_index_ == 0 && index == 0 && size > 14) {
+    /**
+     * BMP file format:
+     * 0-1: Signature (BM)
+     * 2-5: File size
+     * 6-9: Reserved
+     * 10-13: Pixel data offset
+     */
```

Comment:
```suggestion
     *
     * Integer values are stored in little-endian format.
     */
```
Probably obvious to anyone working with BMP, but IMHO it doesn't hurt to point this out ;-)